libcontainer/utils/utils_unix.go

```diff
@@ -23,9 +26,44 @@ func EnsureProcHandle(fh *os.File) error {
 	return nil
 }
 
+var (
+	haveCloseRangeCloexec     bool
+	haveCloseRangeCloexecOnce sync.Once
+)
+
+func tryCloseRangeCloexec(minFd int) error {
+	haveCloseRangeCloexecOnce.Do(func() {
+		// Make sure we're not closing a random file descriptor.
+		tmpFd, err := unix.FcntlInt(0, unix.F_DUPFD_CLOEXEC, 0)
+		if err != nil {
+			return
+		}
+		defer unix.Close(tmpFd)
+
+		err = unix.CloseRange(uint(tmpFd), uint(tmpFd), unix.CLOSE_RANGE_CLOEXEC)
+		// Any error means we cannot use close_range(CLOSE_RANGE_CLOEXEC).
+		// -ENOSYS and -EINVAL ultimately mean we don't have support, but any
+		// other potential error would imply that even the most basic close
+		// operation wouldn't work.
+		haveCloseRangeCloexec = err == nil
+	})
+
+	var err error
+	if haveCloseRangeCloexec {
+		err = unix.CloseRange(uint(minFd), math.MaxUint, unix.CLOSE_RANGE_CLOEXEC)
+	} else {
+		err = unix.ENOSYS
+	}
+	return err
+}
+
```

Comment:
nit: I would rewrite this to
```go
if !haveCloseRangeCloexec {
	return unix.ENOSYS
}
return unix.CloseRange(uint(minFd), math.MaxUint, unix.CLOSE_RANGE_CLOEXEC)
```

or

```go
if haveCloseRangeCloexec {
	return unix.CloseRange(uint(minFd), math.MaxUint, unix.CLOSE_RANGE_CLOEXEC)
}
return unix.ENOSYS
```

for the sake of brevity and readability.